### PR TITLE
Update nightly pipeline to work in dnceng

### PR DESCRIPTION
**Pull request by mthalman**

### CODE REVIEW

File: .vsts-pipelines/builds/dotnet-nightly-size-validation.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 trigger: none
```
Comment (MichaelSimons) [inline .vsts-pipelines/builds/dotnet-nightly-size-validation.yml]: Is this change necessary?  I don't see where this build utilizes any secrets.  It pulls the public images.

File: .vsts-pipelines/steps/cleanup-docker-linux.yml [hunk 2]
```diff
@@ -39,3 +31,16 @@ steps:
...
+################################################################################
+- script: docker rmi -f $(imageNames.imageBuilder.withrepo) 2> /dev/null
+  displayName: Cleanup Image Builder
+  condition: always()
+  continueOnError: true
+
+- script: docker rmi -f $(imageNames.testRunner.withrepo) 2> /dev/null
+  displayName: Cleanup Test Runner
+  condition: always()
+  continueOnError: true
```
Comment (MichaelSimons) [inline .vsts-pipelines/steps/cleanup-docker-linux.yml]: Missing line ending

File: .vsts-pipelines/variables/docker-images.yml [hunk 3]
```diff
@@ -1,4 +1,7 @@
 variables:
   imageNames.imageBuilder.linux: mcr.microsoft.com/dotnet-buildtools/image-builder:debian-20190305214722
   imageNames.imageBuilder.windows: mcr.microsoft.com/dotnet-buildtools/image-builder:nanoserver-20190305134632
+  imageNames.imageBuilder.withrepo: microsoft-dotnet-imagebuilder-withrepo:$(Build.BuildId)-$(System.JobId)
   imageNames.testRunner: mcr.microsoft.com/dotnet-buildtools/prereqs:debian-stretch-docker-testrunner-63f2145-20184325094343
+  imageNames.testRunner.withrepo: microsoft-dotnet-testrunner-withrepo:$(Build.BuildId)-$(System.JobId)
+
```
Comment (MichaelSimons) [inline .vsts-pipelines/variables/docker-images.yml]: extra line with space.

File: .vsts-pipelines/variables/docker-images.yml [hunk 4]
```diff
@@ -1,4 +1,7 @@
 variables:
   imageNames.imageBuilder.linux: mcr.microsoft.com/dotnet-buildtools/image-builder:debian-20190305214722
   imageNames.imageBuilder.windows: mcr.microsoft.com/dotnet-buildtools/image-builder:nanoserver-20190305134632
+  imageNames.imageBuilder.withrepo: microsoft-dotnet-imagebuilder-withrepo:$(Build.BuildId)-$(System.JobId)
```
Comment (MichaelSimons) [inline .vsts-pipelines/variables/docker-images.yml]: Thoughts on dropping `microsoft-dotnet-` from these image names to keep them shorter?  It would help reading the build output.
```suggestion
  imageNames.imageBuilder.withrepo: imagebuilder-withrepo:$(Build.BuildId)-$(System.JobId)
```

File: .vsts-pipelines/variables/docker-images.yml [hunk 5]
```diff
@@ -1,4 +1,7 @@
...
   imageNames.testRunner: mcr.microsoft.com/dotnet-buildtools/prereqs:debian-stretch-docker-testrunner-63f2145-20184325094343
+  imageNames.testRunner.withrepo: microsoft-dotnet-testrunner-withrepo:$(Build.BuildId)-$(System.JobId)
```
Comment (MichaelSimons) [inline .vsts-pipelines/variables/docker-images.yml]:
```suggestion
  imageNames.testRunner.withrepo: testrunner-withrepo:$(Build.BuildId)-$(System.JobId)
```

File: .vsts-pipelines/builds/dotnet-nightly-size-validation.yml (hunk 1 above)
Comment (mthalman) [reply .vsts-pipelines/builds/dotnet-nightly-size-validation.yml]: Ah, nope.  This is left over from my original changes before you had updated the pipelines to use the variable groups.  I had just gone across the board with the updates.  I'll remove it.

File: .vsts-pipelines/variables/docker-images.yml (hunk 4 above)
Comment (mthalman) [reply .vsts-pipelines/variables/docker-images.yml]: Yeah, that makes sense.

File: .vsts-pipelines/steps/cleanup-docker-linux.yml [hunk 6]
```diff
@@ -43,4 +43,5 @@ steps:
 - script: docker rmi -f $(imageNames.testRunner.withrepo) 2> /dev/null
   displayName: Cleanup Test Runner
   condition: always()
-  continueOnError: true
\ No newline at end of file
+  continueOnError: true
+
```
Comment (MichaelSimons) [inline .vsts-pipelines/steps/cleanup-docker-linux.yml]: The line ending here is still messed up.

Comment (mthalman) [reply .vsts-pipelines/steps/cleanup-docker-linux.yml]: Bah.  Fixed
